ambry-api/src/main/java/com/github/ambry/frontend/NamedBlobPath.java

```diff
@@ -73,6 +77,44 @@ public static NamedBlobPath parse(String path, Map<String, Object> args) throws
     }
   }
 
+  /**
+   * Parse the input path for S3 request.
+   * @param path the URI path that needs to be parsed. This path should already be URL decoded and have query paramaters
+   *             removed. In the request handling path, this would be done by the {@link RestRequest} implementation.
+   * @param args a map containing any query parameters from the request.
+   * @return the {@link NamedBlobPath} that indicates the parsing result from blobUrl.
+   * @throws RestServiceException on parsing errors.
+   */
+  public static NamedBlobPath parseS3(String path, Map<String, Object> args) throws RestServiceException {
```

Comment:
Now that we created a new Handler to handle getObjectLockRequest, do we still need this method here? We won't deal with object-lock parameter in NamedBlobPath anymore.

Reply:
We still need this parseS3.
We call parse in many places including accountAndContainerSanityCheck etc. 
We throw exception if we don't have 3 part for list and 4 part for getBlob.